orion-server/src/main/java/com/pinterest/orion/core/actions/memq/MemqTeletraanBrokerReplacementAction.java

```diff
@@ -0,0 +1,143 @@
+package com.pinterest.orion.core.actions.memq;
+
+import com.pinterest.orion.core.actions.generic.NodeAction;
+import com.pinterest.orion.core.memq.MemqCluster;
+import com.pinterest.orion.utils.EC2Helper;
+import com.pinterest.orion.utils.TeletraanClient;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+public abstract class MemqTeletraanBrokerReplacementAction extends NodeAction {
```

Comment:
Is it possible to combine replacement and decommissioning, or make one dependent on the other? I feel the termination part should be the same between the two.

Or this change has to be done at the parent side?

Reply:
I tried to merge the common parts of these two classes into one parent class. It looks ugly. The wait check looks similar but the APIs are different. Plus there're extra steps in replacement workflow. And the existing parent classes are also different. So I rather keep them in two classes and inherant from different parent classes, following existing kafka oop structure.